pkg/image/registry/imagestreamimport/rest.go

```diff
@@ -311,7 +311,7 @@ func checkImportFailure(status api.ImageImportStatus, stream *api.ImageStream, t
 		}
 	}
 
-	if !api.HasTagCondition(stream, tag, condition) {
+	if len(tag) > 0 && !api.HasTagCondition(stream, tag, condition) {
```

Comment:
I'm sorry, one more thing. The conditions will be lost if we don't set them. Could you please log them in some nice way with ERROR level for `len(tag) == 0 && !api.HasTagCondition(stream, tag, condition)`?